CMakeLists.txt

```diff
@@ -51,19 +51,34 @@ add_custom_target(Download)
 add_custom_target(fletch-build-install)
 
 # Options to control GPU support
-option(fletch_BUILD_WITH_CUDA "Build with CUDA support" FALSE)
+option(fletch_ENABLE_CUDA "Build with CUDA support" FALSE)
 
-if (fletch_BUILD_WITH_CUDA)
+if (fletch_ENABLE_CUDA)
   find_package( CUDA QUIET REQUIRED )
 
-  option(fletch_BUILD_WITH_CUDNN "Build with CUDNN support" FALSE)
-
-  if (fletch_BUILD_WITH_CUDNN)
-    set( CUDNN_ROOT "" CACHE PATH "CUDNN root folder" )
-    mark_as_advanced( CUDNN_ROOT )
+  option(fletch_ENABLE_CUDNN "Build with CUDNN support" FALSE)
+  if (fletch_ENABLE_CUDNN)
+    set( CUDNN_ROOT_DIR "" CACHE PATH "CUDNN root folder" )
+    mark_as_advanced( CUDNN_ROOT_DIR )
+    
+    find_library( CUDNN_LIBRARIES
+      NAMES cudnn libcudnn
+      HINTS ${CUDA_TOOLKIT_ROOT_DIR}
```

Comment:
On closer inspection, you are probably doing HINTS right if ${CUDA_TOOLKIT_ROOT_DIR} is coming from the search for CUDA. If so, disregard comment.